### ISSUE TITLE
Issue using vtkboneAIMWriter - Unable to write

### ISSUE THREAD

**Issue by Besler** [Numerics88/vtkbone#2]
The `vtkbone` python package appears unable to write AIM files out. See a test case below and the resulting errors. This has been reproduced on OSX Sierra and OSX El Capitan.

`testAimWrite.py`
```
import vtk
import vtkbone
import os


reader = vtkbone.vtkboneAIMReader()
reader.SetFileName(os.sys.argv[1])
reader.DataOnCellsOff()
print("Reading {}".format(os.sys.argv[1]))
reader.Update()

writer = vtkbone.vtkboneAIMWriter()
writer.SetFileName('temp.aim')
writer.SetInputConnection(reader.GetOutputPort())
writer.SetInputConnection(reader.GetOutputPort())
print('Writing to temp.aim')
writer.Update()
```

Output:
```
$ python testAimWrite.py /Applications/bonelab/data/test25a.aim 
Reading /Applications/bonelab/data/test25a.aim
Writing to temp.aim
libc++abi.dylib: terminating with uncaught exception of type AimIO::AimIOException: Unable to open file 
Abort trap: 6
```

This looks to me like a permissions error. It looks like AimIO cannot open the file `temp.aim` for writing. I am not entirely sure that this is the issue and have not found a fix.

**Comment by Besler** [comment]
The cause of the issue appears to be that `writer.filename` is never set in `vtkboneAIMWriter::WriteData`. As a fix, I propose `writer.filename to this->FileName` after checking if `this->FileName` is valid.

**Comment by Besler** [comment]
A test environment can be created using:
```
conda create -n test -c babesler python=2 vtkbone
source activate test
python
import vtk
import vtkbone
```

The importing above runs successfully. I ran `testAimWrite.py` and it worked successfully. I tried visualizing the data in ParaView and a bunch of errors were thrown. The AIM written from `testAimWrite.py` can be visualized with `vtkboneAIMReader` but cannot be modified with `aimod`. The header can be read with `aix`, however.

I think there is some deeper issue in either `AimIO` or `vtkbone` with writing AIMs, but that is a separate issue.
